dynamiqs/mesolve/rouchon.py

```diff
@@ -115,51 +99,44 @@ def backward_augmented(
         # rho: (b_H, b_rho, n, n) -> (b_H, b_rho, n, n)
         # phi: (b_H, b_rho, n, n) -> (b_H, b_rho, n, n)
 
-        # compute cached operators
-        # H, Hnh, M0, M0rev: (b_H, 1, n, n)
         H = self.H(t)
         Hnh = self.Hnh(H)
-        M0 = self.M0(Hnh)
-        M0rev = self.M0rev(Hnh)
+
+        # === reverse time
+        M0rev, M1srev = self.Ms(Hnh, -self.dt)
```

Comment:
Does this work as intended? I was thinking that `sqrt` of a negative number would raise an error.